### PR TITLE
Add options "reindent_on_save" and "opam_switch"

### PR DESCRIPTION
**Pull request by Pinimo**
Hi! First, thanks for developing the plugin, it was much needed and is very appreciated. 

I suggest adding a few options in the settings file:

- `reindent_on_save` enables deactivating automatic reindent, it might be useful when committing on large files;
- `opam_switch` enables using other opam switches than the default one (just enter the version, e.g. "4.04.1"); in particular, on UNIX, this enables using the config file `$HOME/.ocp/.ocp-indent.conf`. This was indispensable in my job, where we make extensive use of `lwt` (camlp4-like) syntax.

### CODE REVIEW

File: sublime-text-ocp-indent.py [hunk 1]
```diff
@@ -34,7 +40,8 @@ def indent_lines(view, edit, lines, indent_empty = True):
 		stdin = subprocess.PIPE,
 		stdout = subprocess.PIPE,
 		stderr = subprocess.PIPE,
-		universal_newlines = True
+		universal_newlines = True,
+		env = {"PATH":os.getenv("HOME")+"/.opam/"+opam_switch+"/bin/"}
```
Comment (Pinimo) [inline sublime-text-ocp-indent.py]: I don't know if this is supposed to run on Windows... if it is, I think we can build the path with `os.path.join()`.

**Comment by bluepichu** [comment]
Thanks for the contribution!  I'll take a look tomorrow and then we should be able to merge this.

**Comment by Pinimo** [comment]
There was a typo in the global variable name (`reindent_on_saved` -> `reindent_on_save`), I force-pushed to my branch, it should be mergeable. 

However, I am aware one downside of having the path is that only the opam package will ever be used. So maybe it is worth either:

- changing install instructions (install the package on the system switch, or on your favorite switch and edit the preferences file)
- or letting the `opam_switch` parameter empty by default, with a fallback on the `apt` package.

**Comment by bluepichu** [comment]
I think a better option for the environment might be to optionally be able to specify a opam switch, which if set will modify the environment as you have done here, and if unset will leave the environment unmodified (such that the user just needs to have it on their path).

Besides that, I'd try to make sure that the environment stuff is platform-agnostic, and not get rid of the rest of the environment when invoking `ocp-indent` (just in case it's needed for whatever reason).

Together, I think that means the only change from the PR as it currently stands would be to do something like the following around the `subprocess.Popen` call:

```python
# get the proper indentation from ocp-indent
cmd_env = os.environ.copy()

if opam_switch is not None:
	cmd_env["PATH"] = os.path.join(cmd_env["HOME"], ".opam", opam_switch, "bin") + os.pathsep + cmd_env["PATH"]

process = subprocess.Popen(
	command,
	stdin = subprocess.PIPE,
	stdout = subprocess.PIPE,
	stderr = subprocess.PIPE,
	universal_newlines = True,
	env = cmd_env
)
``` 

And to change the default for `opam_switch` in `OcpIndent.sublime-settings` to `null`.

Thoughts?

**Comment by Pinimo** [comment]
Excellent! Thanks. So we come together on this option, indeed best. I'll drop a second commit soon.

**Comment by Pinimo** [comment]
Hello there, after a quiet summer, I made the expected commit. I just tested it on my local, it seems OK.

NB. I personally prefer `reindent_on_save` to be default `false` (sed de gustibus non est disputandum), to avoid git conflicts on bigger files. Should I leave it as is, or change it to false?

**Pull request closed without merging**